File: C++/746 Min Cost Climbing Stairs.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int l = cost.size();
        if (l == 1) return cost[0];
        vector<int> ans(cost.size());
        ans[0] = cost[0];
        ans[1] = cost[1];
        for (int i = 2; i < l; i++) {
            ans[i] = min(ans[i-1], ans[i-2]) + cost[i];
        }
        return min(ans[l-1], ans[l-2]);
    }
};

File: Python/746 Min Cost Climbing Stairs.py
# Min Cost Climbing Stairs
class Solution(object):
    def minCostClimbingStairs(self, cost):
        """
        :type cost: List[int]
        :rtype: int
        """
        ans = [0,0] + [ele for ele in cost]
        for i in range(2, len(cost)):
            ans[i+2] = min(ans[i+1], ans[i]) + cost[i]
        return min(ans[-1], ans[-2])